### PR TITLE
Update NavigationControl when min/maxZoom/Pitch change

### PR DESCRIPTION
**Pull request by rreusser**
See #11013 

`map.transform.minZoom` and `map.transform.maxZoom` affect the appearance of `NavigationControl`, but `map.setMinZoom()` and `map.setMaxZoom()` do not currently notify the control when these values have changed. This PR emits a sequence of dummy events (`zoomstart`, `zoom`, `zoomend`) when the values are set so that the control is updated correctly.

The same goes for min/maxPitch.

Before (note that control does not update to mach minZoom until the map is zoomed):
pre-fix

After (control updates immediately when values are set):
post-fix

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Update NavigationControl when min and max zoom are changed</changelog>`

### CODE REVIEW

File: test/unit/ui/map_events.test.js [hunk 1]
```diff
@@ -532,7 +532,7 @@ test(`Map#on mousedown can have default behavior prevented and still fire subseq
     map.on('click', click);
 
     simulate.click(map.getCanvas());
-    t.ok(click.callCount, 1);
+    t.equal(click.callCount, 1);
```
Comment (rreusser) [inline test/unit/ui/map_events.test.js:535]: A drive-by fix for what appears to be a simple error.

**Review by avpeery — APPROVED** [review]
Nice fix Ricky! LGTM and great unit tests. I don't think this needs a debug page since it seems setMinZoom, etc are already being implemented in different debug pages, but is something to consider.
